packages/flutter_tools/lib/executable.dart

```diff
@@ -59,6 +59,7 @@ import 'src/usage.dart';
 ///
 /// This function is intended to be used from the `flutter` command line tool.
 Future<Null> main(List<String> args) async {
+  SigintProcessSignal.instance.init();
```

Comment:
Ok. We should definitely add a comment that explains this in detail, and ideally a test that verifies this behaviour, because otherwise I guarantee that some well-meaning soul (e.g. me in six months) will "optimise" this.